contribute/testing.md

```diff
@@ -0,0 +1,230 @@
+# Beam Testing Documentation
+
+# Overview
+
+Apache Beam (incubating) is a rapidly-maturing software project with a strong commitment to testing. Consequently, it has many testing-related needs. It requires precommit tests to ensure code going into the repository meets a certain quality bar and it requires ongoing postcommit tests to make sure that more subtle changes which escape precommit are nonetheless caught. This document outlines how to write tests, which tests are appropriate where, and when tests are run, with some additional information about the testing systems at the bottom.
+
+## Beam Java Testing Matrix
+
+
+<table>
```

Comment:
The doc to markdown generator I used generated HTML for the table. Converting it manually would be time- and labor-intensive; if you think it's worth it I can do so but it doesn't seem worthwhile for what we get.